test/ts/index.tsx

```diff
@@ -1,38 +1,49 @@
-import { h, app, ActionsType, View } from "hyperapp"
+import { h, app, View, StateType, UnwiredActions, UnwiredAction } from "hyperapp"
 
 namespace Counter {
-  export interface State {
+  export interface State extends StateType {
     count: number
+    clickCount: number
   }
 
-  export interface Actions {
-    down(): State
-    up(value: number): State
+  export interface Actions extends UnwiredActions<State, Actions> {
+    down: UnwiredAction<State, Actions>
+    up: UnwiredAction<State, Actions, number>
+    addClickCount: UnwiredAction<State, Actions>
   }
 
   export const state: State = {
-    count: 0
+    count: 0,
+    clickCount: 0,
   }
 
-  export const actions: ActionsType<State, Actions> = {
-    down: () => state => ({ count: state.count - 1 }),
-    up: (value: number) => state => ({
-      count: state.count + value
-    })
+  export const actions: Actions = {
+    down: () => (state, actions) => {
+      actions.addClickCount()
+      return { count: state.count - 1 }
+    },
+    up: (value = 1) => (state, actions) => {
+      actions.addClickCount()
+      return { count: state.count + value }
+    },
+    addClickCount: () => state => ({ clickCount: state.clickCount + 1 }),
   }
 }
 
 const view: View<Counter.State, Counter.Actions> = (state, actions) => (
```

Comment:
The `View<...>`type accepts `WiredActions<...>`, but here you pass it `Actions` which are of type `UnwiredActions<...>`, so I have 2 questions:
 - does this actually compiles?
 - inside the view, the user will call an unwiredActions as if it was a wired action? Shouldn't the type of View be:
```typescript
export interface View<
  TState extends StateType,
  TActions extends WiredActions<TState, TActions>
> {
  // Use the proxy here
  (state: TState, actions: ActionProxy<TState, TActions>): VNode<any>
}
```

Other than this, neat changes! With the new API it makes sense now for users to type the unwired actions and get the wired action type auto generated!

Reply:
You're right, it's a bit strange. But it does compile somehow... I'll investigate more :) Thanks for your review 👍

Reply:
You're right. I've changed the signature of the View. But in your snippet the `TActions extends WiredActions<..., ...>` should be `TActions extends UnwiredActions<..., ...>` because you can only proxy UnwiredActions -> WiredActions. To see that it's correct now, you can look at the index.tsx and see what signature the view's `onclick={() => actions.up(5)}`: `(property) up: (data?: any) => Counter.State`. This is correct.

The reason why it compiles is because the UnwiredActions' TActions generic is ambiguous because it's being proxied to WiredActions :')

Reply:
Ah, nice!